docker/mongodb-kubernetes-tests/tests/search/search_enterprise_x509_cluster_auth.py

```diff
@@ -0,0 +1,261 @@
+import yaml
+from kubetester import create_or_update_secret, run_periodically, try_load
+from kubetester.certs import (
+    create_agent_tls_certs,
+    create_tls_certs,
+    create_x509_mongodb_tls_certs,
+)
+from kubetester.kubetester import KubernetesTester
+from kubetester.kubetester import fixture as yaml_fixture
+from kubetester.mongodb import MongoDB
+from kubetester.mongodb_search import MongoDBSearch
+from kubetester.mongodb_user import MongoDBUser
+from kubetester.omtester import skip_if_cloud_manager
+from kubetester.phase import Phase
+from pytest import fixture, mark
+from tests import test_logger
+from tests.common.search import movies_search_helper
+from tests.common.search.search_tester import SearchTester
+from tests.conftest import get_default_operator, get_issuer_ca_filepath
+from tests.search.om_deployment import get_ops_manager
+
+logger = test_logger.get_test_logger(__name__)
+
+ADMIN_USER_NAME = "mdb-admin-user"
+ADMIN_USER_PASSWORD = f"{ADMIN_USER_NAME}-password"
+
+MONGOT_USER_NAME = "search-sync-source"
+MONGOT_USER_PASSWORD = f"{MONGOT_USER_NAME}-password"
+
+USER_NAME = "mdb-user"
+USER_PASSWORD = f"{USER_NAME}-password"
+
+MDB_RESOURCE_NAME = "mdb-ent-tls"
+
+# MongoDBSearch TLS configuration
+MDBS_TLS_SECRET_NAME = "mdbs-tls-secret"
+
+
+@fixture(scope="function")
+def mdb(namespace: str, issuer_ca_configmap: str) -> MongoDB:
+    resource = MongoDB.from_yaml(
+        yaml_fixture("enterprise-replicaset-sample-mflix.yaml"),
+        name=MDB_RESOURCE_NAME,
+        namespace=namespace,
+    )
+
+    if try_load(resource):
+        return resource
+
+    resource.configure(om=get_ops_manager(namespace), project_name=MDB_RESOURCE_NAME)
+    resource.configure_custom_tls(issuer_ca_configmap, "certs")
+    resource["spec"]["security"]["authentication"] = {
+        "enabled": True,
+        "modes": ["X509", "SCRAM"],
+        "agents": {"mode": "X509"},
+        "internalCluster": "X509",
+    }
```

Comment:
This is the only addition to the code already in the `e2e_search_enterprise_tls` test.